homeassistant/helpers/entity_component.py

```diff
@@ -154,6 +160,8 @@ def _async_setup_platform(self, platform_type, platform_config,
         except Exception:  # pylint: disable=broad-except
             self.logger.exception(
                 'Error while setting up platform %s', platform_type)
+        finally:
+            warn_task.cancel()
```

Comment:
Pascal is right. We want to print the warning as soon as 10 seconds are over. If the platform/component *never* finishes, it would never print the warning. We could optionally add another print to add a warning with the total time that a component/platform took.